utils/saasherder.py

```diff
@@ -85,6 +86,23 @@ def _process_template(self, url, path, hash_length, target, parameters):
         resources = oc.process(template, target_parameters)
         return cluster, namespace, resources
 
+    def _collect_images(self, resources):
+        images = set()
+        for resource in resources:
+            try:
+                for c in resource["spec"]["template"]["spec"]["containers"]:
+                    images.add(c["image"])
+            except KeyError:
+                pass
+
+        return images
+
+    def _check_images(self, resources):
+        skopeo = Skopeo()
+        images = self._collect_images(resources)
+        for image in images:
+            skopeo.inspect(image)
```

Comment:
`bool(container.Image(image))` is way cheaper...

Reply:
Also it's more resilient.

Reply:
thanks! and it supports authentication, perfect.